nomad/state/state_store.go

```diff
@@ -357,6 +357,28 @@ func (s *StateStore) UpsertJob(index uint64, job *structs.Job) error {
 		return fmt.Errorf("unable to create job summary: %v", err)
 	}
 
+	// COMPAT 0.4.1 -> 0.5 Create the LocalDisk if it's nil by adding up DiskMB
+	// from task resources
+	for i, tg := range job.TaskGroups {
+		if tg.LocalDisk != nil {
+			continue
```

Comment:
@schmichael No, since having disk resources in the resource block would result in a validation error. So once the user has migrated to a 0.5 setup the system shouldn't have any disk resource asks in the task resource config block.